docs/reference/rest-api/security/query-role.asciidoc

```diff
@@ -0,0 +1,283 @@
+[role="xpack"]
+[[security-api-query-role]]
+=== Query Role API
+
+++++
+<titleabbrev>Query Role</titleabbrev>
+++++
+
+Retrieves roles with <<query-dsl,Query DSL>> in a <<paginate-search-results,paginated>> fashion.
```

Comment:
@albertzaharovits For completeness sake wondering if we should mention that you can't query reserved roles, only native ones somewhere.

Reply:
@n1v0lg Does that look Ok for you: [57675dd](https://github.com/elastic/elasticsearch/pull/110473/commits/57675dd56585773364ebe5456f26d5f3b99701f9)

Reply:
👍 looks great